instrumentation/servlet/src/main/java/brave/servlet/HttpServletHandler.java

```diff
@@ -0,0 +1,62 @@
+package brave.servlet;
+
+import brave.http.HttpAdapter;
+import brave.http.HttpServerHandler;
+import brave.http.HttpServerParser;
+import brave.internal.Nullable;
+import java.nio.ByteBuffer;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import zipkin.Endpoint;
+
+import static brave.internal.InetAddresses.ipStringToBytes;
+
+public final class HttpServletHandler // public for others like sparkjava to use
```

Comment:
@hyleung @llinder so all server impls will need to declare their own HttpAdapter, unless they share types with another library. server impls that have special knowledge of the client address may choose to make their own HttpServerHandler to accomodate that.

I hope looking at the servlet example here might help decide if extending similarly is worth it for ratpack. If you notice gaps, do tell regardless!